libfreerdp/core/aad.c

```diff
@@ -905,44 +900,8 @@ static BOOL read_http_message(rdpAad* aad, BIO* bio, long* status_code, char** c
 
 static BOOL generate_rsa_2048(rdpAad* aad)
 {
-	BOOL rc = FALSE;
 	WINPR_ASSERT(aad);
-
-	EVP_PKEY_CTX* ctx = EVP_PKEY_CTX_new_id(EVP_PKEY_RSA, NULL);
```

Comment:
You need to allocate the `rdpPrivateKey` struct at some point before calling `freerdp_key_generate`

Reply:
done